datalad/metadata/metadata.py

```diff
@@ -27,22 +28,68 @@
 from datalad.interface.results import get_status_dict
 from datalad.interface.utils import eval_results
 from datalad.interface.base import build_doc
+from datalad.metadata.definitions import common_key_defs
 from datalad.support.constraints import EnsureNone
+from datalad.support.constraints import EnsureChoice
 from datalad.support.constraints import EnsureStr
 from datalad.support.gitrepo import GitRepo
 from datalad.support.annexrepo import AnnexRepo
 from datalad.support.param import Parameter
+import datalad.support.ansi_colors as ac
+from datalad.support.json_py import dump as jsondump
+from datalad.support.json_py import load as jsonload
 from datalad.interface.common_opts import recursion_flag
 from datalad.interface.common_opts import recursion_limit
 from datalad.distribution.dataset import Dataset
 from datalad.distribution.dataset import EnsureDataset
 from datalad.distribution.dataset import datasetmethod
 from datalad.utils import unique
+from datalad.utils import assure_list
+from datalad.ui import ui
+from datalad.dochelpers import exc_str
 
 lgr = logging.getLogger('datalad.metadata.metadata')
 
 valid_key = re.compile(r'^[0-9a-z._-]+$')
 
+db_relpath = opj('.datalad', 'metadata', 'dataset.json')
+agginfo_relpath = opj('.datalad', 'metadata', 'aggregate.json')
+
+
+def get_metadata_type(ds, guess=False):
+    """Return the metadata type(s)/scheme(s) of a dataset
+
+    Parameters
+    ----------
+    ds : Dataset
+      Dataset instance to be inspected
+    guess : bool
```

Comment:
Guessing is expensive, as it will happen all the time, not just on aggregate. I really really cannot see a case where the creator of a dataset does not know what kind of metadata it contains, or where a guess would yield a more accurate result. If anything, guessing should be removed completely.